package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -188,4 +189,34 @@ function M.log(message, verbose)
 	posix_syslog.syslog(posix_syslog.LOG_INFO, message)
 end
 
+-- Execute a program found using command PATH search, like the shell.
+-- Return the pid, as well as the I/O streams as pipes or nil on error.
+function M.popen3(path, ...)
+	local r0, w0 = posix_unistd.pipe()
+	local r1, w1 = posix_unistd.pipe()
+	local r2, w2 = posix_unistd.pipe()
+
+	local pid = posix_unistd.fork()
+
+	if pid < 0 then
+		return nil
+	elseif pid == 0 then
+		posix_unistd.close(w0)
+		posix_unistd.close(r1)
+		posix_unistd.close(r2)
+
+		posix_unistd.dup2(r0, posix_unistd.STDIN_FILENO)
+		posix_unistd.dup2(w1, posix_unistd.STDOUT_FILENO)
+		posix_unistd.dup2(w2, posix_unistd.STDERR_FILENO)
```

Comment:
The child process shouldn't get any file descriptors except for 0, 1 and 2. While there is no nice way to ensure this for *all* files that might be opened by the parent process except for looping over all possible FDs, at least `r0`, `w1` and `w2` should be closed after dup2ing them.

Reply:
resolved in 246ed9760184da60ff3b3908afbcce2532cb4578